tests/integ/scala/test_dataframe_reader_suite.py

```diff
@@ -554,13 +554,13 @@ def test_read_parquet_all_data_types_with_no_schema(session, mode):
         StructField('"N"', DecimalType(38, 6), nullable=True),
         StructField('"F"', DoubleType(), nullable=True),
         StructField('"I"', LongType(), nullable=True),
-        StructField('"S"', StringType(), nullable=True),
-        StructField('"C"', StringType(), nullable=True),
+        StructField('"S"', StringType(StringType._MAX_LENGTH), nullable=True),
+        StructField('"C"', StringType(StringType._MAX_LENGTH), nullable=True),
```

Comment:
the schema is inferred from the size of column stored on server side after parquet file is uploaded. In this case, since size is not specified, it is set to `StringType._MAX_LENGTH`. On the client `StringType() != StringType(StringType._MAX_LENGTH)`

Reply:
Can't we make it equivalent by overwriting `__eq__`?